packages/data/src/world/mq/deku_tree_mq.yml

```
@@ -2,126 +2,286 @@
+    "MQ Deku Tree Lobby": "true"
+    "MQ Deku Tree Basement Ledge": "is_adult || event(DEKU_BLOCK) || trick(OOT_DEKU_SKIP) || has_hover_boots || hookshot_anywhere || climb_anywhere"
+    "MQ Deku Tree Room Before Water Room Entry Door": "true"
   events:
-    MQ_DEKU_WATER_TORCHES: "(is_child && has_shield && can_use_sticks) || (can_use_bow && can_play_elegy)"
+    DEKU_MQ_WATER_PATH_TORCH1: "has_fire || can_use_sticks" #You need to go through the next room with fire sticks for this event.
+    DEKU_MQ_BASEMENT_EYE_SWITCH: "can_use_bow || can_use_slingshot"
+  locations:
+    "MQ Deku Tree Basement Chest": "can_use_sticks || has_fire"
+    "MQ Deku Tree Grass Basement Lower 1": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 2": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 3": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 4": "can_cut_grass"
+"MQ Deku Tree Room Before Water Room Entry Door":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Basement": "true"
+    "MQ Deku Tree Room Before Water Room Door to Basement": "event(DEKU_MQ_BASEMENT_EYE_SWITCH)"
+"MQ Deku Tree Room Before Water Room Door to Basement":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room Before Water Room Entry Door": "event(DEKU_MQ_ROOM_BEFORE_WATER_CLEAR)"
+    "MQ Deku Tree Room Before Water Room": "true"
+"MQ Deku Tree Room Before Water Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room Before Water Room Door to Basement": "true"
+    "MQ Deku Tree Room Before Water Room Door to Water Room": "true"
+  events:
+    DEKU_MQ_WATER_PATH_TORCH1: "has_fire" #From xeno: "cond(setting.doorShuffle !== none, has_fire, (roomClearEvent && has_sticks) || has_fire)"
+    DEKU_MQ_ROOM_BEFORE_WATER_CLEAR: "soul_deku_scrub && soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && can_damage"
+  locations:
+    "MQ Deku Tree Grass Room Before Spike 1": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Spike 2": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Spike 3": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Spike 4": "can_cut_grass"
+"MQ Deku Tree Room Before Water Room Door to Water Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room Before Water Room": "true"
+    "MQ Deku Tree Water Room Door to Room Before Water Room": "event(DEKU_MQ_WATER_PATH_TORCH1)"
+"MQ Deku Tree Water Room Door to Room Before Water Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room Before Water Room Door to Water Room": "true"
+    "MQ Deku Tree Water Room Front": "true"
+"MQ Deku Tree Water Room Front":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Water Room Door to Room Before Water Room": "true"
+    "MQ Deku Tree Water Room Back": "(is_child && has_shield) || can_longshot || (can_hookshot && has_iron_boots) || trick(OOT_DEKU_MQ_WATER_ROOM_SPIKE_ROLL)"
+  events:
+    MQ_DEKU_WATER_TORCHES: "can_use_sticks || (can_use_bow && can_play_elegy)" #You can do this as both adult and child, without a shield, by rolling under the spike.
```

```suggestion
   "MQ_DEKU_WATER_TORCHES: "(is_child && has_shield && can_use_sticks) || (can_use_bow && can_play_elegy) || (can_use_sticks && trick(OOT_DEKU_MQ_WATER_ROOM_SPIKE_ROLL))"
```
Also, only adult can longshot over the log. Not sure about the bow over the log though.

---

Fixed.

Just tested; child can longshot just fine over it. Child and bow+elegy works too, just is a little precise (just tested).